dvc_http/__init__.py

```diff
@@ -121,16 +108,24 @@ async def get_client(
             exceptions={aiohttp.ClientError},
         )
 
-        # The default timeout for the aiohttp is 300 seconds
-        # which is too low for DVC's interactions (especially
-        # on the read) when dealing with large data blobs. We
-        # unlimit the total time to read, and only limit the
-        # time that is spent when connecting to the remote server.
+        # The default total timeout for an aiohttp request is 300 seconds
+        # which is too low for DVC's interactions when dealing with large
+        # data blobs. We remove the total timeout, and only limit the time
+        # that is spent when connecting to the remote server and waiting
+        # for new data portions.
+        connect_timeout = kwargs.get("connect_timeout", self.REQUEST_TIMEOUT)
         kwargs["timeout"] = aiohttp.ClientTimeout(
             total=None,
             connect=connect_timeout,
             sock_connect=connect_timeout,
-            sock_read=read_timeout,
+            sock_read=kwargs.get("read_timeout", self.REQUEST_TIMEOUT),
+        )
+
+        kwargs["connector"] = aiohttp.TCPConnector(
+            # Force cleanup of closed SSL transports.
+            # See https://github.com/iterative/dvc/issues/7414
+            enable_cleanup_closed=True,
+            ssl=make_context(kwargs.get("ssl_verify")),

```

Comment:
Looks like we should've pop()-ed ssl_verify here https://github.com/iterative/dvc-http/issues/40

Reply:
Oh, looks like it is already fixed upstream.